libs/vkd3d/device.c

```diff
@@ -2165,6 +2165,65 @@ static void vkd3d_gpu_va_allocator_cleanup(struct vkd3d_gpu_va_allocator *alloca
     pthread_mutex_destroy(&allocator->mutex);
 }
 
+struct d3d12_device_singleton
+{
+    struct list entry;
+    LUID adapter_luid;
+    struct d3d12_device *device;
+};
+
+static pthread_once_t d3d12_device_map_mutex_init_once = PTHREAD_ONCE_INIT;
+static pthread_mutex_t d3d12_device_map_mutex;
+static struct list d3d12_device_map;
+
+static void d3d12_init_device_map_once(void)
+{
+    pthread_mutex_init(&d3d12_device_map_mutex, NULL);
```

Comment:
PTHREAD_MUTEX_INITIALIZER can be used instead.

Reply:
I don't think we have something that maps to Win32 threads for that though

Reply:
We don't have an equivalent of that for Win32 threads so it's not mapped in our thread wrapper iirc.

Reply:
Ok.

Reply:
@Joshua-Ashton There's `SRWLOCK_INIT`, so `#define PTHREAD_MUTEX_INITIALIZER {SRWLOCK_INIT}` should work afaik.